### DISCUSSION TITLE
ID3 meta album art from server -> TFT screen?

### DISCUSSION THREAD

**Discussion by pinter75** [schreibfaul1/ESP32-audioI2S#637] [General]
Great library, I have mp3 files streamed from a server, and can get the MP3 meta info ok. Just wondering if it is also possible to also get and display the album or cover art on a TFT screen?

I see void audio_id3image(File& file, const size_t pos, const size_t size) , can that be used for server streams? I have tested with an embedded image in the mp3 (Front cover art), but the event is not triggered?

Is it that the image is only loaded from an SD card, and not a stream?

**Comment by schreibfaul1** [comment]
After the ID3 header is read and an embedded image is found, the event `audio_id3image()` is thrown.
You will get the information from which position the image starts and how big it is.
Take a look at the project [ESPuino](https://forum.espuino.de/), where the cover image is transferred to a website.

**Reply by pinter75** [reply]
Hi thanks for the reply. I can see that the cover image is read from an MP3 file on SD. If I upload the same file and stream it, the cover is not read?

Ideally I would like to render the cover image from a stream on a TFT.

**Comment by schreibfaul1** [comment]
audio_id3image() just tells you where the image is and how big it is. Then you can read the image with a second SD or SD_MMC instance. At the beginning you will find information about what format it is, e.g. jpeg or png. so you can choose the right decoder.

**Reply by pinter75** [reply]
Audio_id3imge() never finds the image when using a remote streamed MP3, only when local on SD.

Audio.connecttohost("http://host/test.mp3"); = no image

Audio.connecttoSD("test.mp3"); = image found

**Comment by skylai82** [comment]
https://bm.syok.my/home/all
![image](https://github.com/user-attachments/assets/1ee894a4-d3af-4053-9a70-c6bb90606277)
How do I stream this link http://n02.rcs.revma.com/84qe3ve811uvv and get this image displayed to my TFT?
